lib/sqlalchemy/dialects/postgresql/asyncpg.py

```diff
@@ -1060,10 +1086,24 @@ def do_terminate(self, dbapi_connection) -> None:
 
     def create_connect_args(self, url):
         opts = url.translate_connect_args(username="user")
+        multihosts, multiports = self._split_multihost_from_url(url)
 
         opts.update(url.query)
+
+        if multihosts:
+            if not all(multihosts):
+                raise exc.ArgumentError(
+                    "Some hosts is not specified" " in a connection string"
+                )
+            if not all(multiports):
+                raise exc.ArgumentError(
+                    "Some ports is not specified" " in a connection string"
```

Comment:
**mike bayer** (zzzeek) wrote:

messages here should be:  "All hosts are required to be present for asyncpg multiple host URL"  "All ports are required to be present for asyncpg multiple host URL"

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4755